### ISSUE TITLE
RDOC error when installing with gem

### ISSUE THREAD

**Issue by lferro9000** [fastlane/fastlane#11629]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When installing with gem, the installer generates the following error when installing the docs:

```text
Installing darkfish documentation for fastlane-2.77.1
ERROR:  While executing gem ... (RDoc::Error)
    error generating Scan/DetectValues.html: no implicit conversion of nil into String (TypeError)
```

##### Complete output when running fastlane, including the stack trace and command used

Not relevant

### Environment

gem --version
2.7.4

ruby --version
ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]

**Comment by hernangonzalez** [comment]
I get this error everytime! Any idea what this is?

**Comment by lferro9000** [comment]
Still getting this error:

```text
Fetching: fastlane-2.78.0.gem (100%)
Successfully installed fastlane-2.78.0
Parsing documentation for fastlane-2.78.0
Installing ri documentation for fastlane-2.78.0
Installing darkfish documentation for fastlane-2.78.0
ERROR:  While executing gem ... (RDoc::Error)
    error generating Scan/DetectValues.html: no implicit conversion of nil into String (TypeError)
```

**Comment by KrauseFx** [comment]
You didn't tell me how you tried installing fastlane, but you could you try using
```
[sudo] gem install fastlane --no-rdoc --no-ri 
```

**Comment by lferro9000** [comment]
Just with gem update. If installed with no documentation, the issue isn't triggered (as it's related with the documentation only).

I've just run gem update on a different machine (out Jenkins Agent machine), and this was the output:

```text
Parsing documentation for fastlane-2.78.0
Installing ri documentation for fastlane-2.78.0
Installing darkfish documentation for fastlane-2.78.0
Done installing documentation for tty-screen, tty-cursor, tty-spinner, fastlane after 57 seconds
Parsing documentation for fastlane-2.78.0
Parsing documentation for tty-cursor-0.5.0
Parsing documentation for tty-screen-0.6.4
Parsing documentation for tty-spinner-0.8.0
Done installing documentation for fastlane, tty-cursor, tty-screen, tty-spinner after 20 seconds
```
Which is odd as this same version of the documentation on this machine has no such issue.

When i get a bit of time, I will do a comparison of the gems installed and their versions (and also try do the same for the internationalization stuff - aka utf8 lists installed, if i can figure that out).

**Comment by yuki24** [comment]
It looks like the root cause of this issue is actually in RDoc: https://github.com/ruby/rdoc/issues/585, and this is the line that has been affected by the bug:

https://github.com/fastlane/fastlane/blob/620b2bd59b6505c0b7ea49f3c2475dd3c791176e/scan/lib/scan/detect_values.rb#L173-L177

This has already been fixed by https://github.com/ruby/rdoc/pull/586, but there hasn't been any release since it's fixed. For those of you who are affected by this bug, follow @KrauseFx's suggestion above or use Ruby `2.4.3` oreariler to install and run `fastlane`.

**Comment by aycabta** [comment]
I'm RDoc maintainer. I released RDoc 6.0.2 that contains https://github.com/ruby/rdoc/pull/586 two days ago.

```
$ rdoc --version
6.0.2
$ gem install fastlane --version 2.77.1
Fetching: fastlane-2.77.1.gem (100%)
Successfully installed fastlane-2.77.1
Parsing documentation for fastlane-2.77.1
Installing ri documentation for fastlane-2.77.1
Done installing documentation for fastlane after 20 seconds
1 gem installed
```

New Ruby 2.5.x will be released with RDoc 6.0.2 or later (issue: [Ruby core #14617](https://bugs.ruby-lang.org/issues/14617)).

**Comment by joshdholtz** [comment]
@aycabta Thanks for popping in here and letting us know ❤️ Going to close this for now since the possible solutions are using Ruby `2.4.3` or earlier or waiting for official release of Ruby `2.5.x`
